robots/cmd/test-label-analyzer/cmd/filter/matching-tests.go

```diff
@@ -0,0 +1,138 @@
+	testlabelanalyzer "kubevirt.io/project-infra/robots/pkg/test-label-analyzer"
+	"os"
+	"strings"
+)
+
+type filterStatsMatchingTestsOptions struct {
+	outputFilePath string
+
+	outputFile   *os.File
+	inputVersion string
+}
+
+func (o *filterStatsMatchingTestsOptions) validate() error {
+	if o.outputFilePath == "" {
+		return fmt.Errorf("output-file needs to be provided")
+	}
+	_, err := os.Stat(o.outputFilePath)
+	if !os.IsNotExist(err) {
+		return fmt.Errorf("output file %q exists", o.outputFilePath)
+	}
+	o.outputFile, err = os.Create(o.outputFilePath)
+	if err != nil {
+		return fmt.Errorf("could not create output file %q: %v", o.outputFilePath, err)
+	}
+	return nil
+}
+
+var filterStatsMatchingTestsOpts = &filterStatsMatchingTestsOptions{}
+
+var statsMatchingTestsCmd = &cobra.Command{
+	Use:   "matching-tests",
+	Short: "Filters the output of the stats command for matching tests only",
+	Long: `Filters the output of the stats command for matching tests only.
+
+Takes as input the first argument, parses it and converts it into a condensed output that only holds basic test
+information for matching tests, like the name as it would appear inside a Junit XML file or inside a Prow Job run
+overview.
+`,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		if len(args) < 1 {
+			return fmt.Errorf("no input file provided as argument")
+		}
+		return filterMatchingTestsFromInputFile(args[0], filterStatsMatchingTestsOpts)
+	},
+}
+
+func filterMatchingTestsFromInputFile(inputFile string, opts *filterStatsMatchingTestsOptions) error {
+	fileContent, err := os.ReadFile(inputFile)
+	if errors.IsNotFound(err) {
+		return fmt.Errorf("input file %q does not exist", inputFile)
+	} else if err != nil {
+		return fmt.Errorf("input file %q created an error: %v", inputFile, err)
+	}
+	var input testlabelanalyzer.TestFilesStats
+	err = json.Unmarshal(fileContent, &input)
+	if err != nil {
+		return fmt.Errorf("input file %q created an error: %v", inputFile, err)
+	}
+	filtered := filterMatchingTests(input, opts.inputVersion)
+	marshal, err := json.Marshal(filtered)
+	if err != nil {
+		return fmt.Errorf("failed marshalling data: %v", err)
+	}
+	err = os.WriteFile(opts.outputFilePath, marshal, 0666)
+	return err
+}
+
+type matchingTests []matchingTest
+
+type matchingTest struct {
+
+	// Id has the regular expression that matched the test name
+	Id string `json:"id"`
+
+	// Reason has the reason from the rule file that explain why this test is quarantined
+	Reason string `json:"reason"`
+
+	// Version refers to the version this test is quarantined in
+	Version string `json:"version,omitempty"`
+
+	// TestName is the target test name that is matched by the regexp
+	TestName string `json:"test_name"`
+
+	// BlameLine contains the git information about the rule
+	*git.BlameLine `json:"git_blame_line"`
+}
+
+func filterMatchingTests(input testlabelanalyzer.TestFilesStats, inputVersion string) matchingTests {
+	result := make(matchingTests, 0, 0)
```

Comment:
You can just declare an empty slice; no need to specify zero length and zero capacity, like:
```
var results matchingTests
```